cmd/ttn-lw-cli/internal/util/flags.go

```diff
@@ -522,14 +522,10 @@ func setField(rv reflect.Value, path []string, v reflect.Value) error {
 				switch {
 				case ft.AssignableTo(vt):
 					field.Set(v)
-				case ft.Kind() == reflect.Int32 && vt.Kind() == reflect.String:
-					if reflect.PtrTo(ft).Implements(textUnmarshalerType) {
-						err := field.Addr().Interface().(encoding.TextUnmarshaler).UnmarshalText([]byte(v.String()))
-						if err != nil {
-							return err
-						}
-					} else {
-						return fmt.Errorf(`%s does not implement encoding.TextUnmarshaler`, ft.Name())
+				case ft.Kind() == reflect.Int32 && vt.Kind() == reflect.String && reflect.PtrTo(ft).Implements(textUnmarshalerType):
```

Comment:
This `ft.Kind() == reflect.Int32` doesn't look quite right, as that would only make this work for enums and other `int32` types that implement `encoding.TextUnmarshaler`.

Reply:
Originally, I proposed a simple change that works for one specific case which won't affect others. After some review comments, I'm combining cases to make this more efficient. I don't really have much context on the regressions that this change may cause.
Can you provide a final suggestion on what can be done here? Should I just return to #3689 (review)?

Reply:
I think after removing `ft.Kind() == reflect.Int32 &&` this should be fine.